core/conversion/converters/impl/expand.cpp

```diff
@@ -15,24 +16,93 @@ namespace converters {
+    ConversionCtx* ctx,
+    const torch::jit::Node* n,
+    nvinfer1::ITensor* inShape,
+    nvinfer1::ITensor* expandShape,
+    int32_t in_rank,
+    int32_t expand_rank,
+    const char* name) {
+  auto creator = new plugins::CheckShapePluginCreator();
+  std::vector<nvinfer1::PluginField> fields;
+  nvinfer1::PluginField input_rank("input_rank", &in_rank, nvinfer1::PluginFieldType::kINT32, 1);
+  nvinfer1::PluginField output_rank("expand_rank", &expand_rank, nvinfer1::PluginFieldType::kINT32, 1);
+  fields.push_back(input_rank);
+  fields.push_back(output_rank);
+  nvinfer1::PluginFieldCollection collection;
+  collection.nbFields = fields.size();
+  collection.fields = fields.data();
+  auto plugin = creator->createPlugin(name, &collection);
+
+  nvinfer1::ITensor* inputs[] = {inShape, expandShape};
+  auto expandShape_layer = ctx->net->addPluginV2(inputs, 2, *plugin);
+  TRTORCH_CHECK(expandShape_layer, "Unable to create interpolation plugin from node" << *n);
+
+  expandShape_layer->setName("CheckShapePlugin");
+  return expandShape_layer;
+}
+
+void addSliceInput(nvinfer1::Dims& dims, int idx, ConversionCtx* ctx, nvinfer1::ISliceLayer* slice) {
+  int32_t rank = static_cast<int32_t>(dims.nbDims);
+  int32_t* tmp = new int32_t[rank];
+  for (int i = 0; i < rank; i++)
+    tmp[i] = dims.d[i];
+  const nvinfer1::Dims d{1, {rank}};
+  const nvinfer1::Weights w{nvinfer1::DataType::kINT32, tmp, rank};
+  auto t = ctx->net->addConstant(d, w)->getOutput(0);
+  slice->setInput(idx, *t);
+}
+
+nvinfer1::ITensor* vec2Tensor(int32_t* dim, int rank, ConversionCtx* ctx) {
+  const nvinfer1::Dims d{1, {static_cast<int32_t>(rank)}};
+  const nvinfer1::Weights w{nvinfer1::DataType::kINT32, dim, rank};
+  return ctx->net->addConstant(d, w)->getOutput(0);
+}
+
+nvinfer1::ITensor* concat(int max_rank, int old_rank, ConversionCtx* ctx, nvinfer1::ITensor* tensor) {
+  if (max_rank - old_rank > 0) {
+    int32_t* tmp = new int32_t[max_rank - old_rank];
+    for (int i = 0; i < (max_rank - old_rank); i++)
+      tmp[i] = 1;
+    auto max_rank_tensor = vec2Tensor(tmp, max_rank - old_rank, ctx);
+    auto in_shape_tensor = ctx->net->addShape(*tensor)->getOutput(0);
+    nvinfer1::ITensor* const args[2] = {max_rank_tensor, in_shape_tensor};
+    return ctx->net->addConcatenation(args, 2)->getOutput(0);
+  } else { // max_rank - old_rank == 0
+    return ctx->net->addShape(*tensor)->getOutput(0);
+  }
+}
+
 bool add_expand(ConversionCtx* ctx, const torch::jit::Node* n, nvinfer1::ITensor* in, nvinfer1::Dims expandedDims) {
   auto input_dims = in->getDimensions();
   TRTORCH_CHECK(
       input_dims.nbDims <= expandedDims.nbDims,
       "Number of dimensions of the desired expansion must be greater than or equal to the number of input dimensions");
 
   // Validate the expansion. Eg: an input of [3, 1] can be expanded to [1, 3, 4] but not [3, 4, 1]
-  for (int64_t i = expandedDims.nbDims - 1; i >= 0; --i) {
+  for (int i = expandedDims.nbDims - 1; i >= 0; --i) {

```

Comment:
can you change these back to `int64_t`? where ever possible we like to use the explicit sized types.